### ISSUE TITLE
Use of extended character set

### ISSUE THREAD

**Issue by Captainkirkdawson** [FreeUKGen/FreeCENMigration#1258]
Further to my previous comments about some issues with the Welsh Language I wonder if it might be possible to include something specific to Welsh punctuation?

 In Welsh we use a punctuation mark called a ‘To Bach’ which literally translates as ‘little roof’. It is used with the vowels a,e,i,o,u,y and w. (Yes y and w are vowels in Welsh!)

 

So the letters are depicted in certain circumstances as â, ê, î, ô, û, ŷ and ŵ. This punctuation indicates an elongation of the sound so for instance ê is equivalent (almost) to ee.

 

Some of the enumerators I am working on in the 1891 census have used this punctuation mark correctly, but if I transcribe as written I get an error message as the letter typed in the CSV file is not recognised. In modern days we also use a sloping accent on some letters, but this was not the case in the 1800’s so we don’t need to worry about that. However if the ‘To Bach’ letters could be recognised by FC2 I will be able to transcribe what I see correctly.

**Comment by DeniseColbert** [comment]
Pat to have a think/research

**Comment by PatReynolds** [comment]
Welsh uses (https://en.wikipedia.org/wiki/Welsh_orthography#Diacritics):
**circumflex** (used to show a vowel is long, sometimes)
**grave** (used in foreign loans to show that the vowel is short, when Welsh would have it long)
**acute** (used, sometimes, on vowels to indicate stress is on the final syllable,  and to indicate that the word is two syllables rather than one ("e.g. gẃraidd /ˈɡʊ.raið/ (two syllables) "manly", as opposed to gwraidd /ˈɡʷraið/ (one syllable) "root"").
**diaeresi**s (the ") symbol used to show that two vowels are separate, not a dipthong)

These all can be applied to any vowel, ieL a e i o u w y

Implementation
This has links to ucf-8 codes: https://sites.psu.edu/symbolcodes/languages/europe/welsh/  Welsh place names may  need a language tag (cy) - so might need to record them in a separate field?

Possibly of use:
https://historicplacenames.rcahmw.gov.uk/  eg Allt Caer-lêb, Ffos-y-gïach 

National Library of Wales falls over searching Tithe Maps with diacritical marks (PHP error) and this source may not use them.

Could not check in Welsh Marriage Bonds as they are only searchable by personal name.

**Comment by PatReynolds** [comment]
Relevant UCF-8 codes to be activated.

**Comment by Captainkirkdawson** [comment]
Added âêîôûŷŵÂÊÎÔÛŶŴáéíóúýẃÁÉÍÓÝẂäëïöüÿẅÄËÏÖÜŸẄàèìòùẁÀÈÌÒÙẀ as valid characters in broad text fields. Deployed to test3

**Comment by PatReynolds** [comment]
Able to search for Siân (but none found), so that aspect tests fine.

**Comment by PatReynolds** [comment]
@Captainkirkdawson to let @geoffj-FUG know when deployed

**Comment by geoffj-FUG** [comment]
Kirk
Implementation of this change may require a change to the fields types within the ED block of the spreadsheets.
They are currently Narrow Text but if they are to be able to accept Welsh lettering they will need to be amended to Broad Text. 
Fields affected are civil_Parish, ecclesiatical_parish, where_census_taken.
Other possibilities are Ward, parliamentary_constituency and sanitary_district.
Foremane, surname and address are already Broad text.
Geoff

**Comment by Captainkirkdawson** [comment]
@geoffj-FUG I do not believe we should permit their use in the census description fields. To do so would lead to many many problems. 
Their use is intended to permit the TWYS with the dwelling and individual fields of the body of a census document not its description

**Comment by geoffj-FUG** [comment]
Thank you.
I just had to ask.
Geoff

**Comment by geoffj-FUG** [comment]
Kirk

 

Do we need to amend the field type for POBs to Narrow text? Does their being
Broad text create any issues for the future if searching on POBs is
intended? 

 

There will be places with accents etc in their place names. They can be
added to the Gazetteer in the same way as any other Place as necessary. That
should not cause any issues (except for the person keying the entry!). We
already have Gaelic characters in the Irish Gazetteer entries.

 

If so we need to do it now before the other changes take place. We also need
to allow for hyphen (-) to appear in the Place of Birth columns as well as
the Narrow Text characters.

 

Geoff

**Comment by Captainkirkdawson** [comment]
Geoff
I think you are correct to raise this point. There is no doubt that those additional characters would create significant issues for the Gazetteer. Similarly they will create issues for person names with the search; although the latter has more options to retrieve such names with wildcard and soundex.
Perhaps we are creating more problems than actually resolving. I would like the opinion of others. @PatReynolds @rhodamackenzie @AlOneill @FreecenBren

**Comment by rhodamackenzie** [comment]
Is there a simple way to make things work? Perhaps: 
If one searches using any vowel with an accent, all versions of that vowel appear in the results and vice versa, searching without an accent gives results for all variants of that vowel- ie all those with an accent. Or that impossible to code?

**Comment by rhodamackenzie** [comment]
It would be an awful task to add all the alternate names to the Gazetteer. 
For accents, apostrophes and hyphens in the PoB, this would mean a large number of alternates for each place name in the gazetteer. Eg ssss-rrrrr ssss -rrrr ssss - rrr 

And so on and on.

**Comment by Captainkirkdawson** [comment]
But would that be any different from simply having none in the first place. i.e. have we really added anything

**Comment by geoffj-FUG** [comment]
I think that we have 2 options:
1. Make POB Narrow text plus a hyphen. This maintains the status quo. 
We have Gaelic characters  in the Gazetteer because they were imported from the Irish Townlands. They are alternative names to an anglisised name in the main Place name.
2. Keep the current POB of Broad text which will allow accented characters to be entered in the Place Names. However this could result in difficult searching. I am not sure about the complete logic of the search using the Gazetteer but if we can keep the Anglicised name without accents and only allow the accented character in the Other Place name fields we have a tool that to me would be practical.

Looking at the Big Picture, if we ever move to the Irish censuses (which has been designed into the CSVProc spreadsheets and fields) we will need to allow for the accented characters somehow. Some Places will have been enumerated in Gaelic.
We are looking at the issue as Welsh but it is broader than that. 
Rhoda, does the issue affect Scotland in the long run?

Adding accented places to the Gazetteer is no problem if people know how to type them. They would be gradually added at Validation.

Geoff

**Comment by rhodamackenzie** [comment]
So, there are accents in Scottish Gaelic, Irish Gaelic, in Welsh and I wouldn't be surprised if there were a few in English places (eg those with French names) too. 

_There are 2 areas each with 2 choices - inputting data and searching the data._ 

**Inputting, as @geoffj-FUG said above:**
1) no accents allowed but allow hyphens only. Will annoy puritans. Multiple versions of each name would have to be typed into alternative place names to allow for vagaries of spaces before and after a hyphen. 
2) allow anything - these have to be added one at a time. I feel sorry for the Welsh validators!

**Searching:**
1) no accents allowed - inform users that instead of using accents, they need to use wildcards. 
2) allow accents - which will search all placenames and alternative place names but **ONLY** find them if they have been correctly inputted with the correct spacing.

For personal names, I wouldn't allow accents at all, it has now become normalised to not use accents (eg Sian, Zoe) when typing names.

**Comment by PatReynolds** [comment]
The extended character set will only be used for type-what-you-see fields (i.e. personal names and place of birth).  @Captainkirkdawson to investigated the feasabily of using an extendted character set in searching these fields - where a character with a diacrictial mark in a search finds both the character with, and without the mark, and a search for a character without a diacritial mark finds both the character with, and without the mark.  also applies to "-" and " " where a search for one finds the other.

**Comment by geoffj-FUG** [comment]
Pat
I have concerns with
 _and a search for a character without a diacritial mark finds both the character with, and without the mark. also applies to "-" and " " where a search for one finds the other._ 
If this is applied to Validation then mistakes in the enumeration will not be picked up if we ignore the " ". Kingsbrompton for instance would pass testing when the place name is actually Kings Brompton. It would impact on the integrity of the Validation process.
Geoff

**Comment by PatReynoldsFUG** [comment]
Kings Brompton 'missing space' is not a character that would be picked up as acceptable If Kingsbrompton was in the gazzetteer, then it would be acceptable.

For place of birth, this might be important, as for example a Welsh speaker filling in a Welsh return my write his place of birth as Llundain and his wife's as Caer (possibly mutating the inital letters, eg "(in) Chester" becomes (Yng) Nghaer

**Comment by geoffj-FUG** [comment]
Pat

 

We do not enter invalid place name spellings into the Gazetteer. 

 

At Validation the validator will put the correct place spelling into the alternative POB fields. In fact your example is automatically entered in pre-validation.

 

Coordinators used to cheat the system with incorrect spellings in their reference file in FreeCEN1. The Gazetteer reference file is strictly a listing of correct information. 

 

Geoff
